### DISCUSSION TITLE
How can I ensure that the data in a custom object persists when using a new collection?

### DISCUSSION THREAD

**Discussion by GagaPoloJr** [livewire/livewire#7886] [1. Help] [answered]
So, I have a page where there's a live search feature. After the data appears, I can click on an item from the search results to populate the data.

Inside the populate data function, I add a new attribute for the data to be displayed.

The issue is, I have a feature to click increment on each item. After I click, all the public data in Livewire changes. Initially, there were 3 data items, but suddenly it becomes 1, and the collection is lost.

![Screen Recording 2024-02-04 at 08 29 28](https://github.com/livewire/livewire/assets/60550910/32d5a1c8-4c61-43ad-b303-be1de1ba9a58)

**Comment by joshhanley** [comment]
@GagaPoloJr what does your custom object look like? You will need to either make it a `wireable` or create a `synth` to handle them.

https://livewire.laravel.com/docs/properties#wireables
https://livewire.laravel.com/docs/synthesizers

**Reply by GagaPoloJr** [reply]
i think you can forget about the custom object,
maybe i want to ask regarding the livewire, 

 why using query builder not persist if there is an updated properties ?

```php
 $data = Container::with(['resi', 'items'])
            ->join('resi', 'resi.id', '=', 'containers.resi_id')
            ->join('bons', 'bons.container_id', '=', 'containers.id')
            ->join('bongkaran_rinci', 'bongkaran_rinci.container_id', '=', 'containers.id')
            ->where('containers.no_container', $id)
            ->select('bons.*', 'bons.resi_id', 'containers.no_container', 'resi.trip_ke', 'bongkaran_rinci.no_urut_resi')
            ->groupBy('containers.no_container')
            ->get();
```
if there is an updated property the ``` $data``` will return just container model data, not with joined table

if i use eloquent orm like this
```php

  $data = Container::with(['resi', 'items', 'bongkaran_rinci'])
            ->whereHas('bongkaran_rinci', function ($query) use ($id) {
                $query->where('no_container', $id);
            })
            ->select('containers.*')
            ->groupBy('containers.id')
            ->get();

```

the data still persist whether there is an updated property or not

**Reply by joshhanley** [reply, marked as answer]
@GagaPoloJr if you have custom selects (with joins, groupBy, etc) you should put them into a computed property instead of assigning it to a public property as Livewire has no way to know how to restore that data from the database. See here for more details https://livewire.laravel.com/docs/properties#eloquent-constraints-arent-preserved-between-requests

**Reply by GagaPoloJr** [reply]
interesting, I will try with your approach.
 btw it's already solved using orm eloquent.  Thanks !
